src/app/directive/components/directive-edit/directive-edit.component.ts

```diff
@@ -0,0 +1,88 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Directive } from '@models/directive';
+import { AccountService } from '@shared/services/account/account.service';
+import { ApiService } from '@shared/services/api/api.service';
+import { MessageService } from '@shared/services/message/message.service';
+
+@Component({
+  selector: 'pr-directive-edit',
+  templateUrl: './directive-edit.component.html',
+  styleUrls: ['./directive-edit.component.scss'],
+})
+export class DirectiveEditComponent implements OnInit {
+  @Output() public savedDirective = new EventEmitter<Directive>();
+  @Input() public directive: Directive;
+  public email: string;
+  public note: string;
+  public waiting = false;
+  public archiveName: string;
+  public accountExists = true;
+
+  constructor(
+    private api: ApiService,
+    private account: AccountService,
+    private message: MessageService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.directive) {
+      this.email = this.directive.stewardEmail;
+      this.note = this.directive.note;
+    }
+    this.archiveName = this.account.getArchive().fullName;
+  }
+
+  public async submitForm(): Promise<void> {
+    this.waiting = true;
+    this.accountExists = true;
+    try {
+      if (this.directive) {
+        const directive = Object.assign({}, this.directive);
+        directive.stewardEmail = this.email;
+        directive.note = this.note;
+        const response = await this.api.directive.update(directive);
+        this.catchNotFoundError(response);
+        this.directive = response;
+        this.savedDirective.emit(this.directive);
+      } else {
+        const savedDirective = await this.api.directive.create({
+          archiveId: this.account.getArchive().archiveId,
+          type: 'transfer',
+          trigger: {
+            type: 'admin',
+          },
+          stewardEmail: this.email,
+          note: this.note,
+        });
+        this.catchNotFoundError(savedDirective);
+        this.savedDirective.emit(savedDirective);
+      }
+    } catch {
+      if (this.accountExists) {
+        this.message.showError(
+          'There was an error trying to save the Archive Steward information. Please try again.'
+        );
+      }
+    } finally {
+      this.waiting = false;
+    }
+  }
+
+  protected catchNotFoundError(response: any): void {
```

Comment:
I'm mainly using `any` type here to represent that the return value isn't fully known. While the API is supposed to return a `Directive` type, if there is an error (either client-side or sent directly from the back-end) it will return that type instead. I think it might be useful to define the return type of the API as a `Directive | HttpErrorResponse` (or probably just return a `Directive` and throw an `HttpErrorResponse`?), though I think we might want to wait until the actual API is integrated and we can fully test on it. Also ideally I'd like to adjust a lot of the code calling V2 endpoints right now in the web-app before I solidify the types more.

Reply:
If we're trying to represent an unknown return type, would the `unknown` type work?